### ISSUE TITLE
overestimation of large trees' AGB

### ISSUE THREAD

**Issue by cpiponiot** [forestgeo/allodb#120]
One of the main problems with the new version of `allodb`: systemic overestimation of large trees' AGB. The problem seems to stem from our log-log regression that minimizes errors on large values when log-transformed, and from the fact that we don't have enough equations that include DBHs above 100 cm. 

Below is an example showing the resampled values (dots) and the recalibrated equation used in `allodb` (red line), on a natural and log scale. 

![example_overestimation](https://user-images.githubusercontent.com/20208665/106027005-c2099400-60ca-11eb-8181-a8356bafa60f.png)

I'm looking for easily computable ways to calibrate this equation directly on the natural scale, with normally distributed errors (to disandvantage large errors on large trees), with functions such as `nls`. If you have suggestions I'd be very thankful!

**Comment by ValentineHerr** [comment]
I am really not as literate as you with these issues.... 
But, if you want to stay in the natural scale, have you tried `lm(agb ~ poly(dbh, 2, raw = T)` ?
That gives you 3 coefficients (you can call them a, b, and c), 
then your `pred` function becomes 
`pred <- function(x) params$a + params$b *x + params$c * x^2`

and your plot would look like this.
![image](https://user-images.githubusercontent.com/19673377/106031501-5b21b600-609d-11eb-8e6a-3abebad6b34e.png)

**Comment by teixeirak** [comment]
Does this solve it, @cpiponiot ?

**Comment by cpiponiot** [comment]
Thank you @ValentineHerr for the input! 
This would be the easiest function to implement indeed. 
The main problem with the polynomial function however is its behavior for small DBHs, mainly because it doesn't go through the origin (although that can be easily solved) and has a rather inflexible shape. 
I compared the predictions of the nonlinear functions a * x ^b (with the function `nls`) and a*x + b*x^2 (with the function `lm`, similar to what you sent but with an offset so that AGB(dbh= 0) = 0. 

Below are the results: in red the previous predictions, in green a*x^b, and in blue the polynomial function. 

![evaluation](https://user-images.githubusercontent.com/20208665/106115997-eef88e00-6140-11eb-912f-aeaa06174f30.png)

Looking at those results I would be more inclined to use the `nls` version (green) but I haven't used the function very often and I hope it's "stable" enough and won't throw errors in some weird cases (which isn't the case with `lm`). @ValentineHerr do you have more experience with this function?

**Comment by cpiponiot** [comment]
I tried the new method (with) and it seems to be working fine except in a very special case, when only one equation is resampled and it's of the form a*x^b, in which case the resampled data fits the model perfectly and nls throws an error. I found a way around it and it seems to be working fine. 

I added the graphs for all species*site combinations [here](https://github.com/forestgeo/allodb/tree/master/not-package-stuff/graphs/sites) if you want to take a look at it.
